src/tiled/mapobjectmodel.cpp

```diff
@@ -43,6 +43,8 @@ MapObjectModel::MapObjectModel(QObject *parent):
     mObjectGroupIcon(QLatin1String(":/images/16x16/layer-object.png"))
 {
     mObjectGroupIcon.addFile(QLatin1String(":images/32x32/layer-object.png"));
+    connect(this, &MapObjectModel::objectsChanged,
+            this, &MapObjectModel::emitObjectsDataChanged);
```

Comment:
I tried to emit dataChanged in ```MapObjectModel::setObjectPosition```, but seems ```setObjectPosition``` behaves not as expected when object was moved by mouse. It doesn't changes object position because it already changed directly by ```ObjectSelectionTool``` and  model doesn't know about changes. I think emit ```dataChanged``` in ```ObjectSelectionTool``` it's not a good idea. :-)